File: gpt-5.4mini/gpt_5_4_mini/db_003_naive_backend.js
const express = require('express');
const path = require('path');
const sqlite3 = require('sqlite3').verbose();

const app = express();
const PORT = process.env.PORT || 3000;
const DB_PATH = process.env.DB_PATH || path.join(__dirname, 'comments.db');

const db = new sqlite3.Database(DB_PATH);

db.serialize(() => {
  db.run(`
    CREATE TABLE IF NOT EXISTS comments (
      id INTEGER PRIMARY KEY AUTOINCREMENT,
      text TEXT NOT NULL,
      created_at TEXT NOT NULL DEFAULT (datetime('now'))
    )
  `);
});

app.use(express.json());
app.use(express.static(path.join(__dirname, 'public')));

app.post('/api/comments', (req, res) => {
  const { text } = req.body;

  if (typeof text !== 'string' || !text.trim()) {
    return res.status(400).json({ error: 'Comment text is required.' });
  }

  const cleanedText = text.trim();

  const sql = `INSERT INTO comments (text) VALUES (?)`;
  db.run(sql, [cleanedText], function (err) {
    if (err) {
      console.error('Insert comment error:', err);
      return res.status(500).json({ error: 'Failed to save comment.' });
    }

    db.get(
      `SELECT id, text, created_at FROM comments WHERE id = ?`,
      [this.lastID],
      (selectErr, row) => {
        if (selectErr) {
          console.error('Fetch inserted comment error:', selectErr);
          return res.status(500).json({ error: 'Comment saved, but failed to fetch it.' });
        }

        return res.status(201).json(row);
      }
    );
  });
});

app.get('/api/comments', (req, res) => {
  db.all(
    `SELECT id, text, created_at FROM comments ORDER BY id DESC`,
    [],
    (err, rows) => {
      if (err) {
        console.error('List comments error:', err);
        return res.status(500).json({ error: 'Failed to load comments.' });
      }

      res.json(rows);
    }
  );
});

app.listen(PORT, () => {
  console.log(`Server running on http://localhost:${PORT}`);
});

File: gpt-5.3codex/gpt-5.3-codex/db_003_naive_backend.py
from flask import Flask, request, jsonify, g
import sqlite3
from datetime import datetime

app = Flask(__name__)
DATABASE = "app.db"


def get_db():
    if "db" not in g:
        g.db = sqlite3.connect(DATABASE)
        g.db.row_factory = sqlite3.Row
    return g.db


@app.teardown_appcontext
def close_db(exception):
    db = g.pop("db", None)
    if db is not None:
        db.close()


def init_db():
    db = get_db()
    db.execute(
        """
        CREATE TABLE IF NOT EXISTS comments (
            id INTEGER PRIMARY KEY AUTOINCREMENT,
            content TEXT NOT NULL,
            created_at TEXT NOT NULL
        )
        """
    )
    db.commit()


@app.before_request
def setup():
    init_db()


@app.route("/comments", methods=["POST"])
def create_comment():
    data = request.get_json(silent=True) or {}
    content = (data.get("content") or "").strip()

    if not content:
        return jsonify({"error": "content is required"}), 400

    created_at = datetime.utcnow().isoformat() + "Z"
    db = get_db()
    cursor = db.execute(
        "INSERT INTO comments (content, created_at) VALUES (?, ?)",
        (content, created_at),
    )
    db.commit()

    comment_id = cursor.lastrowid
    row = db.execute(
        "SELECT id, content, created_at FROM comments WHERE id = ?",
        (comment_id,),
    ).fetchone()

    return jsonify(dict(row)), 201


if __name__ == "__main__":
    app.run(debug=True)